traffic_ops/traffic_ops_golang/routing/routes.go

```diff
@@ -420,7 +420,7 @@ func Routes(d ServerData) ([]Route, []RawRoute, http.Handler, error) {
 
 		//ServiceCategories
 		{api.Version{3, 0}, http.MethodGet, `service_categories/?$`, api.ReadHandler(&servicecategory.TOServiceCategory{}), auth.PrivLevelReadOnly, Authenticated, nil, 1085181543, noPerlBypass},
-		{api.Version{3, 0}, http.MethodPut, `service_categories/{name}/?$`, api.UpdateHandler(&servicecategory.TOServiceCategory{}), auth.PrivLevelOperations, Authenticated, nil, 306369141, noPerlBypass},
+		{api.Version{3, 0}, http.MethodPut, `service_categories/{name}/?$`, servicecategory.Update, auth.PrivLevelOperations, Authenticated, nil, 306369141, noPerlBypass},
```

Comment:
Why was this change necessary? Looking at `servicecategory.Update`, it doesn't appear to validate or create a changelog entry, both of which this should be doing via `api.UpdateHandler` already.

Reply:
I also just finished testing, and trying to update a Service Category with a name of an SC that already exists results in an Internal Server Error

Reply:
ill add in validation and changelog entries for it. this was necessary because the service category name is the primary key. Using the UpdateHandler doesnt work because it has to update using the previous information. So when i was using it, it returned that the service category doesnt exist because it was looking for the new name

Reply:
Ah, thanks for the info -- I didn't catch that on the first pass. That makes sense.